public/app/plugins/datasource/prometheus/components/monaco-query-field/monaco-completion-provider/situation.ts

```diff
@@ -344,7 +352,11 @@ function resolveLabelMatcherError(node: SyntaxNode, text: string, pos: number):
   }
 
   // now we need to find the other names
-  const otherLabels = getLabels(labelMatchersNode, text);
+  const allLabels = getLabels(labelMatchersNode, text);
+
+  const otherLabels = allLabels.filter((label) => label.name !== labelName);
+
+  // we need to remove "our" label from all-labels, if it is in there
```

Comment:
Does this comment relate to the filtering above? If yes, could we move it just above that line?

Reply:
good catch! fixed.